src/commons/eventsourcing/store.rs

```diff
@@ -453,20 +468,27 @@ where
                         std::process::exit(1);
                     }
 
+                    // Apply events, check that the aggregate can be updated, and make sure
+                    // we have an updated version so we can store it.
                     for event in &events {
-                        self.store_event(event)?;
-
                         agg.apply(event.clone());
-                        if agg.version() % SNAPSHOT_FREQ == 0 {
-                            info.snapshot_version = agg.version();
+                    }
 
-                            self.store_snapshot(&handle, agg)?;
-                        }
+                    // Apply events to sync listeners which may still return errors
+                    for sync_listener in &self.sync_listeners {
+                        sync_listener.as_ref().listen(agg, events.as_slice())?;
                     }
 
+                    // Nothing broke, so it's safe to store the command, events and aggregate
+                    for event in &events {
+                        self.store_event(event)?;
+                    }
+                    info.snapshot_version = agg.version();
+                    self.store_snapshot(&handle, agg)?;
```

Comment:
Before you were only storing a snapshot once every `SNAPSHOT_FREQ` versions, but now you store every time. Is this intentional?

Reply:
Were snapshots costly before because there were lots of events causing snapshots to rise rapidly in size, and now the idea is that less events are saved and so the snapshot is smaller and thus okay to do more often?